packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -480,6 +483,37 @@ const baseDescriptors: PropertyDescriptorMap = ArrayReduce.call(
 
 defineProperties(BaseLightningElement.prototype, baseDescriptors);
 
+// Per Component Constructor, track the corresponding Custom Element
+const ContextProviderMetaMap = new Map();
+
+function getCustomElement(Ctor: ComponentConstructor): HTMLElement {
+    if ((Ctor as any) === BaseLightningElement) {
+        throw new SyntaxError(`Invalid Constructor`);
```

Comment:
we should add to the invalid constructor value to the error message as well as what the expected constructor should be